torch_geometric/utils/get_mp_out.py

```diff
@@ -0,0 +1,49 @@
+from typing import Callable, List
+
+import torch
+from torch import Tensor
+
+from .subgraph import get_num_hops
+
+
+def get_intermediate_messagepassing_embeddings(model: torch.nn.Module,
+                                               **kwargs) -> List[Tensor]:
+    """Get output of all `MessagePassing` layers in
+    :obj:`model`.
+
+    Args:
+        **kwargs: Input to :obj:`model`.
+    """
+    from torch_geometric.nn import MessagePassing
+    intermediate_out = {}
+
+    def get_intermediate_output(name) -> Callable:
```

Comment:
I am not totally convinced we need a nested function here. Why not make `intermediate_out` a list and just append the embeddings?

Reply:
Yeah made `intermediate_out` a list.